src/gt4py/next/program_processors/runners/dace_fieldview/gtir_dataflow.py

```diff
@@ -621,22 +663,53 @@ def _visit_map(self, node: gtir.FunCall) -> DataExpr:
             input_size = input_desc.shape[0]
             if input_size == 1:
                 input_memlets[conn] = dace.Memlet(data=input_node.data, subset="0")
-            elif local_size is not None and input_size != local_size:
-                raise ValueError(f"Invalid node {node}")
+            elif input_size != local_size:
+                raise ValueError(
+                    f"Argument to map node with local size {input_size}, expected {local_size}."
+                )
             else:
                 input_memlets[conn] = dace.Memlet(data=input_node.data, subset=map_index)
-                local_size = input_size
+                # filter input expressions with skip values and store their connectivity
+                input_local_offset_provider = self.subgraph_builder.get_offset_provider(
+                    input_expr.local_offset
+                )
+                assert isinstance(input_local_offset_provider, gtx_common.Connectivity)
+                if input_local_offset_provider.has_skip_values:
+                    skip_value_connectivities.append(
+                        (input_expr.local_offset, input_local_offset_provider)
+                    )
 
             input_nodes[input_node.data] = input_node
 
-        if local_size is None:
-            # corner case where map is applied to 1-element lists
-            assert len(input_nodes) >= 1
-            local_size = 1
-
         out, _ = self.sdfg.add_temp_transient((local_size,), dtype)
         out_node = self.state.add_access(out)
 
+        if len(skip_value_connectivities) != 0:
+            # In case one or more of input expressions contain skip values, we use
+            # the connectivity-based offset provider as mask for map computation.
+            local_offset, offset_provider = skip_value_connectivities[0]
```

Comment:
What if you have multiple?

Reply:
But stuff is added to `skip_value_connectivities` if there are skip values.
The concrete value is irrelevant, only the fact that they are present matters.